### PR TITLE
Battery simulation for m5paper in mcsim

### PR DESCRIPTION
**Pull request by FWeinb**
Similar to #827 added support for the battery in mcsim. Demo:

https://user-images.githubusercontent.com/1250430/152782825-beb104ab-ef05-49dd-816c-7a556eb4b697.mov

### CODE REVIEW

**Comment by phoddie** [comment]
Excellent. The battery is really valuable in the simulator, because testing it in real-time on the device takes **forever**. Thank you.

File: build/simulators/m5paper/peripherals.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  constructor(options) {
+    super(options);
+    this.#config = { time: undefined, enabled: true };
+    this.postJSON({
+      rtc: this.#config,
+    });
+  }
+
+  enabled() {
+    return this.#config.enabeld;
```
Comment (phoddie) [inline build/simulators/m5paper/peripherals.js]: typo – should be `enabled`

**Comment by phoddie** [comment]
If I'm reading the code correctly, it looks like you also have enough of an RTC simulation in place that a project can get /set the RTC, though the RTC time doesn't progress beyond the time set. That is useful for developing set-up code.

**Comment by phoddie** [comment]
Regarding `read` on the battery, the returned value should be from 0 to 1, so that it is independent of the particular battery. The implementation of `read` would be:

```js
  read() {
    return (this.#voltage - 3300) / 1000;
  }
```

If the voltage is useful, that be a separate call.

File: build/simulators/m5paper/peripherals.js (hunk 1 above)
Comment (FWeinb) [reply build/simulators/m5paper/peripherals.js]: Thanks.

**Comment by FWeinb** [comment]
> If I'm reading the code correctly, it looks like you also have enough of an RTC simulation in place that a project can get /set the RTC, though the RTC time doesn't progress beyond the time set. That is useful for developing set-up code.

I played with a RTC Simulation, forgot to remove this stuff in this PR. If you want I can add that to the UI. 

> Regarding read on the battery, the returned value should be from 0 to 1

I mirrored the implementation of the battery from here: 
https://github.com/Moddable-OpenSource/moddable/blob/4e39861fe5f00b5006393a0b42f84b64f17b5ea0/build/devices/esp32/targets/m5paper/host/provider.js#L197-L209

that one also give back the voltage.

**Comment by phoddie** [comment]
So then, it was my mistake. Apologies for not noticing sooner. It is kind of great that implementing the simulator revealed a problem in the device code thought. ;) That being the case, we should update both to return a value from 0 to 1.

**Comment by FWeinb** [comment]
I fixed the simulator and the m5paper implementation.

**Comment by phoddie** [comment]
Much appreciated. We'll get that merged.

**Comment by phoddie** [comment]
Merged. I also updated battery `read` on the device to read several samples and average, as the reading was quite noisy in my tests. I built a small app to see that. It is in [this gist](https://gist.github.com/phoddie/c7b8bf4917e163a7bc3ff3ab96ca949c), FWIW.

**Pull request closed without merging**